backend/src/main/kotlin/blue/mild/covid/vaxx/service/VaccinationSlotService.kt

```diff
@@ -0,0 +1,127 @@
+package blue.mild.covid.vaxx.service
+
+import blue.mild.covid.vaxx.dao.model.EntityId
+import blue.mild.covid.vaxx.dao.model.Locations
+import blue.mild.covid.vaxx.dao.model.VaccinationSlots
+import blue.mild.covid.vaxx.dao.repository.LocationRepository
+import blue.mild.covid.vaxx.dao.repository.VaccinationSlotRepository
+import blue.mild.covid.vaxx.dto.request.CreateVaccinationSlotsDtoIn
+import blue.mild.covid.vaxx.dto.request.query.VaccinationSlotStatus
+import blue.mild.covid.vaxx.dto.response.VaccinationSlotDtoOut
+import blue.mild.covid.vaxx.error.entityNotFound
+import mu.KLogging
+import org.jetbrains.exposed.sql.Column
+import org.jetbrains.exposed.sql.Op
+import org.jetbrains.exposed.sql.and
+import java.time.Instant
+
+@Suppress("MagicNumber")
+val MAX_FUTURE = Instant.now().plusMillis(2*365*24*60*60*1000)
+val DEFAULT_STATUS = VaccinationSlotStatus.ONLY_FREE
+
+class VaccinationSlotService(
+    private val locationRepository: LocationRepository,
+    private val vaccinationSlotRepository: VaccinationSlotRepository,
+//    private val patientRepository: PatientRepository
+) {
+    private companion object : KLogging()
+
+    @Suppress("TooGenericExceptionThrown", "ThrowsCount")
+    suspend fun addSlots(createDto: CreateVaccinationSlotsDtoIn, locationId: EntityId? = null): List<EntityId> {
+        if (createDto.to < createDto.from.plusMillis(createDto.durationMillis)) {
+            throw Exception("Specified time range is not valid - ${createDto}")
+        }
+
+        if (createDto.locationId != null && locationId != null) {
+            throw Exception("Location mismatch - ${createDto} vs ${locationId}")
+        }
+        val usedLocationId = (createDto.locationId ?: locationId)!!
```

Comment:
Please use `requireNotNull(createDto.locationId ?: locationId) { "A message why this should not happen" }` if this scenario happens (even when it should not), it gives us a reasonable error message in the logs.

Reply:
Fixed